backend/clients/perplexity_client.py

```
@@ -0,0 +1,41 @@
+                api_key: Perplexity API key. If not provided, uses PERPLEXITY_API_KEY env variable.
+            """
+            self.api_key = api_key or os.environ.get("PERPLEXITY_API_KEY")
+            if not self.api_key:
+                logger.warning("Perplexity API key not provided. Research enhancement will be disabled.")
+            else:
+                self.client = OpenAI(api_key=self.api_key, base_url="https://api.perplexity.ai")
+                logger.debug(f"Initialized OpenAI client with Perplexity base URL")
+            
+        # Define the function to run in the executor
+        def call_perplexity(self, messages: list[dict]):
+            print(f"[DIRECT PRINT] Inside executor: Making Perplexity API call")
```

Ruff (0.8.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)